lib/std/math.zig

```diff
@@ -1122,6 +1122,38 @@ pub fn isPowerOfTwo(v: anytype) bool {
     return (v & (v - 1)) == 0;
 }
 
+/// Aligns the given integer type bit width to a width divisible by 8.
+pub fn ByteAlignedInt(comptime T: type) type {
+    const info = @typeInfo(T).Int;
+    if (info.bits == 0) return std.meta.Int(info.signedness, 8);
+    const bits = (info.bits + 7) / 8 * 8;
+    const extended_type = std.meta.Int(info.signedness, bits);
+    return extended_type;
+}
+
+/// Aligns the given integer bit width to a width divisible by 8.
+pub fn byteAlignInt(value: anytype) ByteAlignedInt(@TypeOf(value)) {
+    return @intCast(ByteAlignedInt(@TypeOf(value)), value);
+}
+
+test "byteAlignInt" {
+    try testing.expect(byteAlignInt(@as(u0, 0)) == @as(u8, 0));
+    try testing.expect(byteAlignInt(@as(i0, 0)) == @as(i8, 0));
+    try testing.expect(byteAlignInt(@as(u3, 4)) == @as(u8, 4));
+    try testing.expect(byteAlignInt(@as(u8, 255)) == @as(u8, 255));
+    try testing.expect(byteAlignInt(@as(i111, -(1 << 100))) == @as(i112, -(1 << 100)));
+    try testing.expect(byteAlignInt(@as(u129, (1 << 128))) == @as(u136, (1 << 128)));
+}
+
+test "ByteAlignedInt" {
+    try testing.expect(ByteAlignedInt(u0) == u8);
```

Comment:
Why does `u0` turn into `u8`?

Reply:
Because it contains only one value - 0. So it can be placed in 1 byte. This is how I thought about it

Reply:
Yes but it doesn't need to be since it can only contain one value.

Reply:
Does that mean that it shouldn't be written by `Writer`?

Reply:
Yes.